### ISSUE TITLE
Git is showing me some strange errors?

### ISSUE THREAD

**Issue by KurtE** [ROBOTIS-GIT/OpenCM9.04#32]
Not sure if this is the right place for this and/or is this an issue with my local machine and/or with your git project. 

When I tried to Sync up to the develop branch, either just my simple clone of your project or if I try to fetch upstream for my fork/clone to update to recent stuff.   I have found that github for windows will appear to hang and when I did the fetch upstream from command prompt it took a long long time and then gave me this message:
```
D:\GitHub\OpenCM9.04>git fetch upstream
remote: Counting objects: 48, done.
remote: Compressing objects: 100% (32/32), done.
remote: Total 48 (delta 15), reused 43 (delta 11), pack-reused 0
Unpacking objects: 100% (48/48), done.
error: cannot lock ref 'refs/remotes/upstream/develop': is at 3e82af723dbbd4373600e6c3c4b9fe96056a1313 but expected 2d052eb45614d6467a722e13a46fd43bb4c11aaa
From https://github.com/ROBOTIS-GIT/OpenCM9.04
 ! 2d052eb..3e82af7  develop    -> upstream/develop  (unable to update local ref)
error: cannot lock ref 'refs/remotes/upstream/master': is at 4ffa401488ba90f9940e90452375d23f3a5a6828 but expected 13d8087a3cb5693c2a01c63e0f1a3bfeca1b34b5
 ! 13d8087..4ffa401  master     -> upstream/master  (unable to update local ref)
 * [new tag]         1.2.0      -> 1.2.0

D:\GitHub\OpenCM9.04>
```
And I unfortunately only know GIT well enough to be dangerous ;) 

Suggestions?

**Comment by OpusK** [comment]
Hi, @KurtE 

Please refer to the following two documents and try again.
- [Configuring a remote for a fork](https://help.github.com/articles/configuring-a-remote-for-a-fork/)
- [Syncing a fork](https://help.github.com/articles/syncing-a-fork/)

**Comment by KurtE** [comment]
Hi @OpusK,

I am actually reasonably familiar with the basics.  It is just recently on this project, I have been seeing some strange messages, like I mentioned.   My sequence for syncing the fork is a little different than is shown in the link you mentioned.    At times wondering if maybe it is or was issues with Github for Windows (new version came out yesterday after this).

Here is a run this morning using command line that did not give this error, but gives a different warning, which may be related:

```
D:\GitHub>cd OpenCM9.04

D:\GitHub\OpenCM9.04>git remote -v
origin  https://github.com/KurtE/OpenCM9.04.git (fetch)
origin  https://github.com/KurtE/OpenCM9.04.git (push)
upstream        https://github.com/ROBOTIS-GIT/OpenCM9.04.git (fetch)
upstream        https://github.com/ROBOTIS-GIT/OpenCM9.04.git (push)

D:\GitHub\OpenCM9.04>git pull
Already up-to-date.

D:\GitHub\OpenCM9.04>git fetch

D:\GitHub\OpenCM9.04>git checkout develop
Already on 'develop'
Your branch is up-to-date with 'origin/develop'.

D:\GitHub\OpenCM9.04>git fetch upstream
remote: Counting objects: 24, done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 24 (delta 4), reused 24 (delta 4), pack-reused 0
Unpacking objects: 100% (24/24), done.
From https://github.com/ROBOTIS-GIT/OpenCM9.04
   3e82af7..1e74c88  develop    -> upstream/develop

D:\GitHub\OpenCM9.04>git reset --hard upstream/develop
HEAD is now at 1e74c88 Fixed #33

D:\GitHub\OpenCM9.04>git push origin develop --force
Counting objects: 24, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (21/21), done.
Writing objects: 100% (24/24), 5.75 MiB | 414.00 KiB/s, done.
Total 24 (delta 17), reused 0 (delta 0)
remote: Resolving deltas: 100% (17/17), completed with 14 local objects.
remote: warning: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
remote: warning: See http://git.io/iEPt8g for more information.
remote: warning: File arduino/opencm_arduino/opencm9.04_release/variants/OpenCM904/lib_f103.a is 66.79 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
To https://github.com/KurtE/OpenCM9.04.git
   3e82af7..1e74c88  develop -> develop
```

Will see if it repeats again or not... I am guessing that maybe the github for windows failed to download  that file properly (again I have seen errors) and so the updates put up the wrong version, which then the next download gave the above error?

Note: the reason I like this method of syncing my fork, is it does not leave any new artificial git entries on my fork... That is after I did the above, the git log last entry is your Fixed #33 message

**Comment by OpusK** [comment]
The last warning seems to be a capacity problem with the library (.a) file.
Perhaps you have a problem with your local repository, and it seems that the previous error has occurred.

Please let me know if any of these things happen again :)

**Comment by KurtE** [comment]
Thanks, 

 

My guess is that there was an issue with GIT probably related with github for windows.   When I saw this issue, I saw problems both with my simple clone of your project which I keep in a separate location, so I can see what has changed, as well as my Fork/clone.  Where it would try to compress the data from the server, then sort of hang for a long time in the 90+% range and then give me some form of message like I mentioned. 

 

I was able to just now sync up my simple clone to your last change (Serial.flush()) and it went right through without messages or delay. 

 

Again it is probably a GIT issue, that hopefully was resolved.  You probably should  just close this issue and see if it pops up again. 

 

Kurt
